### DISCUSSION TITLE
How to use offlineExchange to only store specific query ?

### DISCUSSION THREAD

**Discussion by kai-you2** [urql-graphql/urql#3278] [Q&A] [answered]
Hi, I tried to use `offlineExchange`  to store some of the massive data but not easily changed to the indexedDB, and it works well, but I found out that **I can't seem to set my `offlineExchange` to only store specific query for those data**, since the other data and query  didn't want to be store in indexedDB, I don't seem to see the related config in `offlineExchange`

Did anyone know how to acheive the behavior ?

One of the idea I came up with is to **write a custom exchange to determine different query and store the data**, but I think the offlineExchange should have these kinds of config for us to use ?

**Comment by kitten** [comment, marked as answer]
You can write a custom storage to not persist some of the data, or change the `requestPolicy` per query. However, partial caching or partial storage isn't part of Graphcache.

Even if it would be, a custom storage makes more sense in these cases. While how the data is stored is technically an implementation detail, the problem with allowing this to be customised anyway is that the data will be cached either way until your application restarts, so there's little point in turning this into a specific API.
